### PR TITLE
calculator class based

### PR DESCRIPTION
**Pull request by ogiogiovictor**
# Changes

1. Completed the Class based function 
2. Implement the calculate function

### CODE REVIEW

File: src/components/calculator/calculator.component.jsx [hunk 1]
```diff
@@ -1,56 +1,76 @@
...
-                { this.state.map(btn => (
-               <Button key={btn.text} text={btn.text} type={btn.type} />
-            ) )}
+                <Screen info={next || total || '0'} opr={operation} tx={total} onChange={(e) => this.showContent(e.target.value)} />
+
+                <div class='calculator-grid'>
+                    {button.map(btn => (
+                        <Button key={btn.text} text={btn.text} type={btn.type} act={this.act} />
+                    ))}
+                </div>
```
Comment (Mre55) [inline src/components/calculator/calculator.component.jsx]: Kindly, make sure to display the operator when doing the calculation. The user have to know what operator they choose. Please implement that.

https://user-images.githubusercontent.com/34578631/177872648-cd8460c0-490f-47aa-92cb-a4f297d47b7f.mov

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @ogiogiovictor,**

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The README is well documented ✅ 
- Linters are working properly ✅ 

## Required Changes ♻️🛠️ 
_Check the comments under the review._




## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator/calculator.component.jsx (hunk 1 above)
Comment (ogiogiovictor) [reply src/components/calculator/calculator.component.jsx]: Dear Mihreteab,

Thank you for the feedback. 
Please find below the project requirement. The math operator was not included in the requirement. 

image

Comment (Mre55) [reply src/components/calculator/calculator.component.jsx]: @ogiogiovictor, On the last bullet point it say test your app in the browser and should work as expected. The operation works fine but the only problem is that the clicked operator is not displayed on the browser as of the numbers. I hope that clarify the point.

Comment (ogiogiovictor) [reply src/components/calculator/calculator.component.jsx]: Dear Mihreteab,

Thank you for the feedback, however, the only said it should be able to perform math operation which the application is already doing.

Comment (Whoistolu) [reply src/components/calculator/calculator.component.jsx]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @ogiogiovictor 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**